chainerx_cc/chainerx/routines/math.cc

```diff
@@ -554,6 +554,26 @@ Array LogSumExp(const Array& x, const OptionalAxes& axis, bool keepdims) {
 
 Array LogSoftmax(const Array& x, const OptionalAxes& axis) { return x - LogSumExp(x, axis.has_value() ? axis : OptionalAxes{1}, true); }
 
+Array Square(const Array& x) {
+    Array out = EmptyLike(x, x.device());
+
+    {
+        NoBackpropModeScope scope{};
+        x.device().Square(x, out);
+    }
+
+    BackwardBuilder bb{"square", x, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([x_tok = bb.RetainInput(0)](BackwardContext& bctx) {
+            const Array& x = bctx.GetRetainedInput(x_tok);
+            bctx.input_grad() = *bctx.output_grad() * (2 * x);
```

Comment:
A nitpick and optional for you to apply but the parenthesis here feels a bit verbose.
```suggestion
            bctx.input_grad() = *bctx.output_grad() * 2 * x;
```